### PR TITLE
OEL-167: Add plugin tests

### PR DESCRIPTION
**Pull request by claudiu-cristea**

### CODE REVIEW

File: tests/src/Kernel/BackendTest.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use Drupal\search_api\Entity\Index;
+use Drupal\search_api\Entity\Server;
+use Drupal\search_api\Utility\Utility;
+use Drupal\Tests\search_api\Functional\ExampleContentTrait;
+
+/**
+ * Class for testing Europa Search drupal integration.
+ */
+class BackendTest extends KernelTestBase {
+  use ExampleContentTrait;
```
Comment (claudiu-cristea) [inline tests/src/Kernel/BackendTest.php:26]: Needs a new line before.

File: tests/src/Kernel/BackendTest.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
```
Comment (claudiu-cristea) [inline tests/src/Kernel/BackendTest.php]: Spaces surrounding =

File: tests/src/Kernel/BackendTest.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\search_api\Entity\Index;
+use Drupal\search_api\Entity\Server;
+use Drupal\search_api\Utility\Utility;
+use Drupal\Tests\search_api\Functional\ExampleContentTrait;
+
+/**
+ * Class for testing Europa Search drupal integration.
+ */
```
Comment (claudiu-cristea) [inline tests/src/Kernel/BackendTest.php]: Needs a `@group` tag

File: tests/src/Kernel/BackendTest.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+namespace Drupal\oe_search\Tests;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\search_api\Entity\Index;
+use Drupal\search_api\Entity\Server;
+use Drupal\search_api\Utility\Utility;
+use Drupal\Tests\search_api\Functional\ExampleContentTrait;
+
+/**
+ * Class for testing Europa Search drupal integration.
```
Comment (claudiu-cristea) [inline tests/src/Kernel/BackendTest.php]: Better description?

**Review by claudiu-cristea — COMMENTED** [review]
This is an older review with all points already fixed.

File: tests/src/Kernel/BackendTest.php [hunk 5]
```diff
@@ -124,7 +127,50 @@ public function testIndexItems(): void {
...
+    $entity = $item->getOriginalObject()->getValue();
+    // Assert query parameters.
+    parse_str($request->getUri()->getQuery(), $parameters);
+    $this->assertSame($entity->toUrl()->setAbsolute()->toString(), $parameters['uri']);
+    $this->assertSame($this->getSiteHash() . '-europa_search_index-' . $item_id, $parameters['reference']);
+    $this->assertSame('["en"]', $parameters['language']);
+    // Assert request body.
+    $this->inspectBoundary($request);
+    $parts = $this->getMultiParts($request);
+    $expected_meta = json_encode([
```
Comment (claudiu-cristea) [inline tests/src/Kernel/BackendTest.php:220]: Missing `"ext-json": "*"` from `require-dev`

File: tests/src/Kernel/BackendTest.php [hunk 6]
```diff
@@ -157,4 +203,31 @@ protected function assertServiceMockCalls(string $path, int $applies_calls, int
...
+    return $requests[$path];
+  }
+
+  /**
+   * Returns the saved hash for the current site.
+   *
+   * @return string
+   *   The site hash.
+   */
+  protected function getSiteHash(): string {
```
Comment (claudiu-cristea) [inline tests/src/Kernel/BackendTest.php]: So, maybe it makes sense now to move `SearchApiEuropaSearchBackend::getSiteHash()` in a utility class?

Comment (drishu) [reply tests/src/Kernel/BackendTest.php]: yes

File: composer.json [hunk 7]
```diff
@@ -11,13 +11,14 @@
         "drupal/core-composer-scaffold": "^8.9 || ^9.1",
         "drupal/search_api": "^1.19",
         "http-interop/http-factory-guzzle": "^1.0",
-        "openeuropa/europa-search-client": "dev-EPIC-EuropaSearch",
+        "openeuropa/europa-search-client": "dev-OEL-167",
```
Comment (claudiu-cristea) [inline composer.json]: Upstream merged. Should revert this change.

Comment (claudiu-cristea) [reply composer.json]: Already fixed by me.
